loopy/target/c/codegen/expression.py

```diff
@@ -743,4 +743,65 @@ def map_array_literal(self, expr, enclosing_prec):
 
 # }}}
 
+
+class ExpressionToCVecExpressionMapper(ExpressionToCExpressionMapper):
```

Comment:
Are `ExpressionToCVecExpressionMapper ` and `CVecExpressionToCodeMapper` not going to be specialised anymore? As in we would go through `ExpressionToCExpressionMapper ` and `CExpressionToCodeMapper `  and special case there? Or do we keep these as is?